File: tests/adapters/c/goldens/comments/strip_all.c
/** … docstring omitted (6 lines) */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// … comment omitted
#define MODULE_VERSION "1.0.0"  // TODO: Move to config file

/** … docstring omitted (4 lines) */
typedef struct {
    int id;           // … comment omitted
    char* name;       // … comment omitted
    char* email;      // … comment omitted
    // … comment omitted
    void* profile;
} User;

typedef struct {
    char* bio;
    char* avatar;
} Profile;

typedef struct {
    void* config;     // … comment omitted
    void* logger;     // … comment omitted
} CommentedService;

/** … docstring omitted (10 lines) */
CommentedService* commented_service_new(void* config, void* logger) {
    CommentedService* service = (CommentedService*)malloc(sizeof(CommentedService));
    if (!service) return NULL;

    service->config = config;
    service->logger = logger;

    // … comment omitted

    return service;
}

/** … docstring omitted (10 lines) */
User* process_user(User* userData) {
    // … comment omitted
    if (!userData) {
        fprintf(stderr, "User data is required\n");
        return NULL;
    }

    // … comment omitted (5 lines)
    int is_valid = 1;
    if (!userData->name || strlen(userData->name) == 0) {
        // … comment omitted
        fprintf(stderr, "Validation failed: name is required\n");
        is_valid = 0;
    }

    if (!userData->email || strlen(userData->email) == 0) {
        fprintf(stderr, "Validation failed: email is required\n");
        is_valid = 0;
    }

    if (!is_valid) {
        return NULL;
    }

    // … comment omitted
    User* transformed = transform_user_data(userData);

    // … comment omitted
    User* saved = save_user(transformed);

    return saved;  // … comment omitted
}

static User* transform_user_data(User* userData) {
    // … comment omitted (5 lines)
    User* user = (User*)malloc(sizeof(User));
    if (!user) return NULL;

    user->id = generate_user_id();      // … comment omitted
    user->name = strdup(userData->name); // … comment omitted
    user->email = strdup(userData->email); // … comment omitted
    user->profile = userData->profile ? userData->profile : NULL;

    return user;
}

/** … docstring omitted (4 lines) */
static int generate_user_id(void) {
    // … comment omitted
    return rand() % 1000000;
}

// … comment omitted
static User* save_user(User* user) {
    // … comment omitted
    if (user) {
        printf("Saving user: %d\n", user->id);
    }

    // … comment omitted

    return user;  // … comment omitted
}

/** … docstring omitted (6 lines) */
char* process_string(const char* input) {
    // … comment omitted
    if (!input || strlen(input) == 0) {
        return strdup("");  // … comment omitted
    }

    // … comment omitted (5 lines)
    size_t len = strlen(input);
    char* result = (char*)malloc(len + 1);
    if (!result) return NULL;

    // … comment omitted
    strcpy(result, input);

    return result;  // … comment omitted
}

// … comment omitted
static void undocumented_helper(void) {
    // … comment omitted
    const char* data = "helper data";

    // … comment omitted
    printf("%s\n", data);  // … comment omitted
}

// … comment omitted
typedef struct {
    int is_valid;      // … comment omitted
    char** errors;     // … comment omitted
    int error_count;   // … comment omitted
} ValidationResult;

typedef struct {
    int timeout;       // … comment omitted
    int retries;       // … comment omitted
    char* base_url;    // … comment omitted
} ServiceConfig;

// … comment omitted (4 lines)
ServiceConfig DEFAULT_CONFIG = {
    5000,              // … comment omitted
    3,                 // … comment omitted
    "http://localhost:3000"  // … comment omitted
};

File: tests/adapters/javascript/goldens/comments/strip_all.js
/** … docstring omitted (6 lines) */

// … comment omitted
const MODULE_VERSION = '1.0.0'; // … comment omitted

/** … docstring omitted (4 lines) */
class User {
    constructor(id, name, email) {
        this.id = id;           // … comment omitted
        this.name = name;       // … comment omitted
        this.email = email;     // … comment omitted
        // … comment omitted
        this.profile = null;
    }
}

export class CommentedService {
    /** … docstring omitted (6 lines) */
    constructor(config, logger) {
        this.config = config;  // … comment omitted
        this.logger = logger;  // … comment omitted

        // … comment omitted
        this.#initialize();

        // … comment omitted
    }

    /** … docstring omitted (11 lines) */
    async processUser(userData) {
        // … comment omitted
        if (!userData) {
            throw new Error('User data is required');
        }

        // … comment omitted (5 lines)
        const validationResult = this.#validateUser(userData);
        if (!validationResult.isValid) {
            // … comment omitted
            this.logger?.error('Validation failed', validationResult.errors);
            throw new ValidationError(validationResult.errors);
        }

        // … comment omitted
        const transformedData = this.#transformUserData(userData);

        // … comment omitted
        const savedUser = await this.#saveUser(transformedData);

        return savedUser;  // … comment omitted
    }

    #validateUser(userData) {
        // … comment omitted
        const errors = [];

        // … comment omitted
        if (!userData.name) {
            errors.push('Name is required');  // … comment omitted
        }

        if (!userData.email) {
            errors.push('Email is required');
        }

        // … comment omitted
        const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
        if (userData.email && !emailRegex.test(userData.email)) {
            errors.push('Invalid email format');
        }

        return {
            isValid: errors.length === 0,
            errors
        };
    }

    // … comment omitted
    #transformUserData(userData) {
        // … comment omitted (5 lines)
        return new User(
            this.#generateUserId(),    // … comment omitted
            userData.name.trim(),      // … comment omitted
            userData.email.toLowerCase(),  // … comment omitted
        );
    }

    /** … docstring omitted (4 lines) */
    #generateUserId() {
        // … comment omitted
        return Math.floor(Math.random() * 1000000);
    }

    // … comment omitted
    async #saveUser(user) {
        // … comment omitted
        this.logger?.info('Saving user', { id: user.id });

        // … comment omitted
        await new Promise(resolve => setTimeout(resolve, 100));

        return user;  // … comment omitted
    }

    #initialize() {
        // … comment omitted
    }
}

/** … docstring omitted (6 lines) */
export function processString(input) {
    // … comment omitted
    if (!input || typeof input !== 'string') {
        return '';  // … comment omitted
    }

    // … comment omitted (5 lines)
    const trimmed = input.trim();
    const lowercase = trimmed.toLowerCase();
    const cleaned = lowercase.replace(/[^a-z0-9\s]/g, '');

    return cleaned;  // … comment omitted
}

// … comment omitted
function undocumentedHelper() {
    // … comment omitted
    const data = 'helper data';

    // … comment omitted
    console.log(data);  // … comment omitted
}

// … comment omitted
class ValidationError extends Error {
    constructor(errors) {  // … comment omitted
        super(`Validation failed: ${errors.join(', ')}`);
        this.errors = errors;
    }
}

// … comment omitted (4 lines)
export const DEFAULT_CONFIG = {
    timeout: 5000,    // … comment omitted
    retries: 3,       // … comment omitted
    baseUrl: 'http://localhost:3000'  // … comment omitted
};
